spyder_kernels/console/shell.py

```diff
@@ -64,10 +64,10 @@ def __init__(self, *args, **kwargs):
 
     def ask_exit(self):
         """Engage the exit actions."""
-        if self.active_eventloop != "inline":
+        if self.active_eventloop not in [None, "inline"]:
             # Some eventloops prevent the kernel from shutting down
             self.enable_gui('inline')
-        return super(SpyderShell, self).ask_exit()
+        super(SpyderShell, self).ask_exit()
```

Comment:
```suggestion
        super().ask_exit()
```

Use `super` in the way allowed by Python 3.